tests/main/auto-refresh-pre-download/task.yaml

```diff
@@ -0,0 +1,109 @@
+summary: Check that an inhibited auto-refresh triggers a pre-download change and resumes on close
+
+# Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+systems: [-ubuntu-14.04-*]
+
+kill-timeout: 5m
+
+environment:
+  REFRESH_TYPE/close: "close"
+  REFRESH_TYPE/ignore: "ignore-running"
+
+prepare: |
+  snap install --devmode jq
+  snap install test-snapd-sh
+  snap refresh --unhold
+
+restore: |
+  snap remove --purge jq || true
+  snap remove --purge test-snapd-sh || true
+
+debug: |
+  snap changes
+  cat /home/test/notif.log || true
+
+execute: |
+  changeAfterID() {
```

Comment:
Given `changeAfterID` have a lot in common with `snapd-state` `_wait_autorefresh()`, I'm
tempted to propose the addition of a more generic `snapd-state` `wait_for_change` that also handles testing for unexpected changes. Perhaps more appropriate when if is more widely useful.